healthcheck/healthcheck.go

```diff
@@ -150,20 +150,52 @@ type Observation struct {
 	// Backoff is the amount of time we back off between retries for failed
 	// checks.
 	Backoff time.Duration
+
+	// OnSuccess is a callback which will be executed when the healthcheck
+	// succeeds. This is optional.
+	OnSuccess func()
+
+	// OnFailure is a callback which will be executed when the healthcheck
+	// fails. This is optional.
+	OnFailure func()
+}
+
+type ObservationOption func(*Observation)
```

Comment:
nit: all public/exported types and functions need Godoc comments.

Reply:
✅